framework/web/Response.php

```diff
@@ -547,23 +547,29 @@ public function sendStreamAsFile($handle, $attachmentName, $options = [])
 
     /**
      * Sets a default set of HTTP headers for file downloading purpose.
-     * @param string $attachmentName the attachment file name
+	 * @param string $attachmentName the file name shown to the user. Must be specified in UTF-8 encoding. If null,
+	 * the browser will try to determine the filename from the URL.
      * @param string $mimeType the MIME type for the response. If null, `Content-Type` header will NOT be set.
      * @param boolean $inline whether the browser should open the file within the browser window. Defaults to false,
      * meaning a download dialog will pop up.
      * @param integer $contentLength the byte length of the file being downloaded. If null, `Content-Length` header will NOT be set.
      * @return $this the response object itself
      */
-    public function setDownloadHeaders($attachmentName, $mimeType = null, $inline = false, $contentLength = null)
+    public function setDownloadHeaders($attachmentName = null, $mimeType = null, $inline = false, $contentLength = null)
     {
         $headers = $this->getHeaders();
 
         $disposition = $inline ? 'inline' : 'attachment';
         $headers->setDefault('Pragma', 'public')
             ->setDefault('Accept-Ranges', 'bytes')
             ->setDefault('Expires', '0')
-            ->setDefault('Cache-Control', 'must-revalidate, post-check=0, pre-check=0')
-            ->setDefault('Content-Disposition', "$disposition; filename=\"$attachmentName\"");
+            ->setDefault('Cache-Control', 'must-revalidate, post-check=0, pre-check=0');
+
+		if($attachmentName !== null) {
+			$headers->setDefault('Content-Disposition', $disposition . '; filename*=UTF-8\'\'' . rawurlencode($attachmentName));
```

Comment:
Older browsers aren't recognizing `filename*` at all so `filename` should be always provided.